backend/src/services/lectures/lectures.routes.js

```diff
@@ -0,0 +1,287 @@
+            message: 'You are not enrolled in this course',
+          });
+        }
+
+        // Check if course exists
+        const courseExists = await lecturesRepo.courseExists(courseId);
+        if (!courseExists) {
+          return reply.code(404).send({ error: 'Course not found' });
+        }
+
+        // Get all lectures
+        const lectures = await lecturesRepo.getLecturesByCourseId(courseId);
+        return reply.send({ lectures });
+      } catch (error) {
+        return mapAndReply(error, reply);
+      }
+    }
+  );
+
+  // Get a single lecture
+  fastify.get(
+    '/api/courses/:course_id/lectures/:lecture_id',
+    {
+      preHandler: fastify.authenticate,
+      schema: lecturesSchemas.GetLectureSchema,
+    },
+    async (req, reply) => {
+      try {
+        const courseId = parseInt(req.params.course_id, 10);
+        const lectureId = parseInt(req.params.lecture_id, 10);
+
+        if (isNaN(courseId) || isNaN(lectureId)) {
+          return reply.code(400).send({
+            error: 'Invalid course_id or lecture_id',
+          });
+        }
+
+        const userId = req.user.id;
+
+        // Check if user can view lectures
+        const canView = await lecturesPermissions.canViewLectures(
+          userId,
+          courseId
+        );
+        if (!canView) {
+          return reply.code(403).send({
+            error: 'Forbidden',
+            message: 'You are not enrolled in this course',
+          });
+        }
+
+        // Get lecture
+        const lecture = await lecturesRepo.getLectureById(lectureId, courseId);
+        if (!lecture) {
+          return reply.code(404).send({ error: 'Lecture not found' });
+        }
+
+        return reply.send({ lecture });
+      } catch (error) {
+        return mapAndReply(error, reply);
+      }
+    }
+  );
+
+  // Create a new lecture
+  fastify.post(
+    '/api/courses/:course_id/lectures',
+    {
+      preHandler: fastify.authenticate,
+      schema: lecturesSchemas.CreateLectureSchema,
+    },
+    async (req, reply) => {
+      try {
+        const courseId = parseInt(req.params.course_id, 10);
+        if (isNaN(courseId)) {
+          return reply.code(400).send({ error: 'Invalid course_id' });
+        }
+
+        const userId = req.user.id;
+
+        // Check if user can modify lectures
+        const canModify = await lecturesPermissions.canModifyLectures(
+          userId,
+          courseId
+        );
+        if (!canModify) {
+          return reply.code(403).send({
+            error: 'Forbidden',
+            message: 'Only professors and TAs can create lectures',
+          });
+        }
+
+        // Check if course exists
+        const courseExists = await lecturesRepo.courseExists(courseId);
+        if (!courseExists) {
+          return reply.code(404).send({ error: 'Course not found' });
+        }
+
+        // Validate lecture data
+        const validation = lecturesService.validateLectureData(req.body);
```

Comment:
[Suggestion]
With the schema definitions in place, these validations to confirm the type of data should not be needed. Can you see if removing this method still sends errors back with invalid data?